hal/property_manager.c

```diff
@@ -426,23 +426,36 @@ int get_property(const char* prop, char* data, size_t max_len) {
 }
 
 int get_channel_for_path( const char *path ) {
+	int r;
+
+	PRINT( VERBOSE,"%s(): %s\n", __func__, NULL == path ? "(null)" : path );
 
 	if ( NULL == path ) {
-		return -1;
+		r = -1;
+		goto out;
 	}
 	if ( strlen( path ) < 4 ) {
-		return -1;
+		r = -1;
+		goto out;
 	}
 	if (
-		! (
-			0
-			|| 0 == strncmp( "rx/", path, 2 )
-			|| 0 == strncmp( "tx/", path, 2 )
-		)
+		1
+		&& 0 != strncmp( "rx/", path, 3 )
+		&& 0 != strncmp( "tx/", path, 3 )
+		&& 0 != strncmp( "rx_", path, 3 )
+		&& 0 != strncmp( "tx_", path, 3 )
 	) {
-		return -1;
+		PRINT( ERROR,"%s(): %s does not begin with {rt}x{_/}\n", __func__, NULL == path ? "(null)" : path );
+		r = -1;
+		goto out;
 	}
-	return 'a' - path[ 3 ];
+
+	r = path[ 3 ] - 'a';
+
+	PRINT( VERBOSE,"%s(): %s => %d\n", __func__, NULL == path ? "(null)" : path, r );
+
+out:
```

Comment:
It isn't being used for debug purposes. I'm not even sure how that is possible.

If you've been trained to believe that goto is bad based on Dijkstra's opinion (when he was rather young and foolish), you should read his follow-up publication where he more or less apologizes for his original paper.

Goto makes perfect sense for system-level software, which is exactly what Dijkstra's argument was saying it should be restricted to.

In any case, I just though that I should point that out. I'll remove the goto statement, but strictly speaking, it's absolutely fine to keep it there.